Compilation error with ESP IDF 5.1.1

I used the previous version with ESP IDF 5.1.1 and had no problem. After updating to 1.1.0, I get these error messages:

> esp_lcd_st7796.c:77:29: error: 'esp_lcd_panel_dev_config_t' has no member named 'rgb_ele_order'
> esp_lcd_st7796.c:78:10: error: 'LCD_RGB_ELEMENT_ORDER_RGB' undeclared (first use in this function)

I checked esp_lcd_panel_dev_config_t in ESP IDF and in fact it doesn't have such a member.

--- a/components/lcd/esp_lcd_gc9503/idf_component.yml
+++ b/components/lcd/esp_lcd_gc9503/idf_component.yml
@@ -1,8 +1,8 @@
-version: "2.1.0"
+version: "3.0.0"
 targets:
   - esp32s3
 description: ESP LCD GC9503
 url: https://github.com/espressif/esp-bsp/tree/master/components/lcd/esp_lcd_gc9503
 dependencies:
-  idf: ">=5.0.4"
+  idf: ">5.0.4,!=5.1.1"
   cmake_utilities: "0.*"

--- a/components/lcd/esp_lcd_gc9503/test_apps/main/idf_component.yml
+++ b/components/lcd/esp_lcd_gc9503/test_apps/main/idf_component.yml
@@ -1,6 +1,6 @@
 ## IDF Component Manager Manifest File
 dependencies:
-  idf: ">=5.0.0"
+  idf: ">5.0.4,!=5.1.1"
   esp_lcd_panel_io_additions:
     version: "^1"
     public: true

--- a/components/lcd/esp_lcd_gc9a01/README.md
+++ b/components/lcd/esp_lcd_gc9a01/README.md
@@ -22,7 +22,8 @@ Alternatively, you can create `idf_component.yml`. More is in [Espressif's docum
 
 ```c
     ESP_LOGI(TAG, "Initialize SPI bus");
-    const spi_bus_config_t bus_config = GC9A01_PANEL_BUS_SPI_CONFIG(EXAMPLE_PIN_NUM_LCD_PCLK, EXAMPLE_PIN_NUM_LCD_MOSI);
+    const spi_bus_config_t bus_config = GC9A01_PANEL_BUS_SPI_CONFIG(EXAMPLE_PIN_NUM_LCD_PCLK, EXAMPLE_PIN_NUM_LCD_MOSI,
+                                                                    EXAMPLE_LCD_H_RES * 80 * sizeof(uint16_t));
     ESP_ERROR_CHECK(spi_bus_initialize(EXAMPLE_LCD_HOST, &bus_config, SPI_DMA_CH_AUTO));
 
     ESP_LOGI(TAG, "Install panel IO");
@@ -51,18 +52,22 @@ Alternatively, you can create `idf_component.yml`. More is in [Espressif's docum
     // };
     const esp_lcd_panel_dev_config_t panel_config = {
         .reset_gpio_num = EXAMPLE_PIN_NUM_LCD_RST,      // Set to -1 if not use
-#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 4)      // Implemented by LCD command `36h`
-        .color_space = ESP_LCD_COLOR_SPACE_BGR,
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
+        .color_space = ESP_LCD_COLOR_SPACE_RGB,
 #else
-        .rgb_ele_order = LCD_RGB_ELEMENT_ORDER_BGR,
+        .rgb_endian = LCD_RGB_ENDIAN_RGB,
 #endif
-        .bits_per_pixel = EXAMPLE_LCD_BIT_PER_PIXEL,    // Implemented by LCD command `3Ah` (16/18)
+        .bits_per_pixel = 16,                           // Implemented by LCD command `3Ah` (16/18)
         // .vendor_config = &vendor_config,            // Uncomment this line if use custom initialization commands
     };
     ESP_ERROR_CHECK(esp_lcd_new_panel_gc9a01(io_handle, &panel_config, &panel_handle));
     ESP_ERROR_CHECK(esp_lcd_panel_reset(panel_handle));
     ESP_ERROR_CHECK(esp_lcd_panel_init(panel_handle));
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
+    ESP_ERROR_CHECK(esp_lcd_panel_disp_off(panel_handle, false));
+#else
     ESP_ERROR_CHECK(esp_lcd_panel_disp_on_off(panel_handle, true));
+#endif
 ```
 
 There is an example in ESP-IDF with this LCD controller. Please follow this [link](https://github.com/espressif/esp-idf/tree/master/examples/peripherals/lcd/spi_lcd_touch).

--- a/components/lcd/esp_lcd_gc9a01/esp_lcd_gc9a01.c
+++ b/components/lcd/esp_lcd_gc9a01/esp_lcd_gc9a01.c
@@ -61,7 +61,7 @@ esp_err_t esp_lcd_new_panel_gc9a01(const esp_lcd_panel_io_handle_t io, const esp
         ESP_GOTO_ON_ERROR(gpio_config(&io_conf), err, TAG, "configure GPIO for RST line failed");
     }
 
-#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 4)
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
     switch (panel_dev_config->color_space) {
     case ESP_LCD_COLOR_SPACE_RGB:
         gc9a01->madctl_val = 0;
@@ -74,15 +74,15 @@ esp_err_t esp_lcd_new_panel_gc9a01(const esp_lcd_panel_io_handle_t io, const esp
         break;
     }
 #else
-    switch (panel_dev_config->rgb_ele_order) {
-    case LCD_RGB_ELEMENT_ORDER_RGB:
+    switch (panel_dev_config->rgb_endian) {
+    case LCD_RGB_ENDIAN_RGB:
         gc9a01->madctl_val = 0;
         break;
-    case LCD_RGB_ELEMENT_ORDER_BGR:
+    case LCD_RGB_ENDIAN_BGR:
         gc9a01->madctl_val |= LCD_CMD_BGR_BIT;
         break;
     default:
-        ESP_GOTO_ON_FALSE(false, ESP_ERR_NOT_SUPPORTED, err, TAG, "unsupported color element order");
+        ESP_GOTO_ON_FALSE(false, ESP_ERR_NOT_SUPPORTED, err, TAG, "unsupported rgb endian");
         break;
     }
 #endif

--- a/components/lcd/esp_lcd_gc9a01/idf_component.yml
+++ b/components/lcd/esp_lcd_gc9a01/idf_component.yml
@@ -1,4 +1,4 @@
-version: "1.1.0"
+version: "1.2.0"
 description: ESP LCD GC9A01
 url: https://github.com/espressif/esp-bsp/tree/master/components/lcd/esp_lcd_gc9a01
 dependencies:

--- a/components/lcd/esp_lcd_gc9a01/include/esp_lcd_gc9a01.h
+++ b/components/lcd/esp_lcd_gc9a01/include/esp_lcd_gc9a01.h
@@ -62,16 +62,17 @@ esp_err_t esp_lcd_new_panel_gc9a01(const esp_lcd_panel_io_handle_t io, const esp
  *
  * @param[in] sclk SPI clock pin number
  * @param[in] mosi SPI MOSI pin number
+ * @param[in] max_trans_sz Maximum transfer size in bytes
  *
  */
-#define GC9A01_PANEL_BUS_SPI_CONFIG(sclk, mosi)         \
-    {                                                   \
-        .sclk_io_num = sclk,                            \
-        .mosi_io_num = mosi,                            \
-        .miso_io_num = -1,                              \
-        .quadhd_io_num = -1,                            \
-        .quadwp_io_num = -1,                            \
-        .max_transfer_sz = SPI_LL_DMA_MAX_BIT_LEN >> 3, \
+#define GC9A01_PANEL_BUS_SPI_CONFIG(sclk, mosi, max_trans_sz)   \
+    {                                                           \
+        .sclk_io_num = sclk,                                    \
+        .mosi_io_num = mosi,                                    \
+        .miso_io_num = -1,                                      \
+        .quadhd_io_num = -1,                                    \
+        .quadwp_io_num = -1,                                    \
+        .max_transfer_sz = max_trans_sz,                        \
     }
 
 /**

--- a/components/lcd/esp_lcd_gc9a01/test_apps/main/test_esp_lcd_gc9a01.c
+++ b/components/lcd/esp_lcd_gc9a01/test_apps/main/test_esp_lcd_gc9a01.c
@@ -14,6 +14,7 @@
 #include "esp_heap_caps.h"
 #include "esp_log.h"
 #include "esp_lcd_panel_io_interface.h"
+#include "esp_lcd_panel_io.h"
 #include "esp_lcd_panel_ops.h"
 #include "unity.h"
 #include "unity_test_runner.h"
@@ -70,7 +71,8 @@ static void test_draw_bitmap(esp_lcd_panel_handle_t panel_handle)
 TEST_CASE("test gc9a01 to draw color bar with SPI interface", "[gc9a01][spi]")
 {
     ESP_LOGI(TAG, "Initialize SPI bus");
-    const spi_bus_config_t buscfg = GC9A01_PANEL_BUS_SPI_CONFIG(TEST_PIN_NUM_LCD_PCLK, TEST_PIN_NUM_LCD_DATA0);
+    const spi_bus_config_t buscfg = GC9A01_PANEL_BUS_SPI_CONFIG(TEST_PIN_NUM_LCD_PCLK, TEST_PIN_NUM_LCD_DATA0,
+                                    TEST_LCD_H_RES * 80 * TEST_LCD_BIT_PER_PIXEL / 8);
     TEST_ESP_OK(spi_bus_initialize(TEST_LCD_HOST, &buscfg, SPI_DMA_CH_AUTO));
 
     ESP_LOGI(TAG, "Install panel IO");
@@ -84,10 +86,10 @@ TEST_CASE("test gc9a01 to draw color bar with SPI interface", "[gc9a01][spi]")
     esp_lcd_panel_handle_t panel_handle = NULL;
     const esp_lcd_panel_dev_config_t panel_config = {
         .reset_gpio_num = TEST_PIN_NUM_LCD_RST,
-#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 4)
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
         .color_space = ESP_LCD_COLOR_SPACE_BGR,
 #else
-        .rgb_ele_order = LCD_RGB_ELEMENT_ORDER_BGR,
+        .rgb_endian = LCD_RGB_ENDIAN_BGR,
 #endif
         .bits_per_pixel = TEST_LCD_BIT_PER_PIXEL,
     };
@@ -96,10 +98,10 @@ TEST_CASE("test gc9a01 to draw color bar with SPI interface", "[gc9a01][spi]")
     TEST_ESP_OK(esp_lcd_panel_init(panel_handle));
     TEST_ESP_OK(esp_lcd_panel_invert_color(panel_handle, true));
     TEST_ESP_OK(esp_lcd_panel_mirror(panel_handle, true, false));
-#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
-    TEST_ESP_OK(esp_lcd_panel_disp_on_off(panel_handle, true));
-#else
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
     TEST_ESP_OK(esp_lcd_panel_disp_off(panel_handle, false));
+#else
+    TEST_ESP_OK(esp_lcd_panel_disp_on_off(panel_handle, true));
 #endif
 
     test_draw_bitmap(panel_handle);

--- a/components/lcd/esp_lcd_ili9341/README.md
+++ b/components/lcd/esp_lcd_ili9341/README.md
@@ -22,7 +22,8 @@ Alternatively, you can create `idf_component.yml`. More is in [Espressif's docum
 
 ```c
     ESP_LOGI(TAG, "Initialize SPI bus");
-    const spi_bus_config_t bus_config = ILI9341_PANEL_BUS_SPI_CONFIG(EXAMPLE_PIN_NUM_LCD_PCLK, EXAMPLE_PIN_NUM_LCD_MOSI);
+    const spi_bus_config_t bus_config = ILI9341_PANEL_BUS_SPI_CONFIG(EXAMPLE_PIN_NUM_LCD_PCLK, EXAMPLE_PIN_NUM_LCD_MOSI,
+                                                                     EXAMPLE_LCD_H_RES * 80 * sizeof(uint16_t));
     ESP_ERROR_CHECK(spi_bus_initialize(EXAMPLE_LCD_HOST, &bus_config, SPI_DMA_CH_AUTO));
 
     ESP_LOGI(TAG, "Install panel IO");
@@ -51,18 +52,22 @@ Alternatively, you can create `idf_component.yml`. More is in [Espressif's docum
     // };
     const esp_lcd_panel_dev_config_t panel_config = {
         .reset_gpio_num = EXAMPLE_PIN_NUM_LCD_RST,      // Set to -1 if not use
-#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 4)      // Implemented by LCD command `36h`
-        .color_space = ESP_LCD_COLOR_SPACE_BGR,
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)      // Implemented by LCD command `36h`
+        .color_space = ESP_LCD_COLOR_SPACE_RGB,
 #else
-        .rgb_ele_order = LCD_RGB_ELEMENT_ORDER_BGR,
+        .rgb_endian = LCD_RGB_ENDIAN_RGB,
 #endif
-        .bits_per_pixel = EXAMPLE_LCD_BIT_PER_PIXEL,    // Implemented by LCD command `3Ah` (16/18)
+        .bits_per_pixel = 16,                           // Implemented by LCD command `3Ah` (16/18)
         // .vendor_config = &vendor_config,            // Uncomment this line if use custom initialization commands
     };
     ESP_ERROR_CHECK(esp_lcd_new_panel_ili9341(io_handle, &panel_config, &panel_handle));
     ESP_ERROR_CHECK(esp_lcd_panel_reset(panel_handle));
     ESP_ERROR_CHECK(esp_lcd_panel_init(panel_handle));
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
+    ESP_ERROR_CHECK(esp_lcd_panel_disp_off(panel_handle, false));
+#else
     ESP_ERROR_CHECK(esp_lcd_panel_disp_on_off(panel_handle, true));
+#endif
 ```
 
 There is an example in ESP-IDF with this LCD controller. Please follow this [link](https://github.com/espressif/esp-idf/tree/master/examples/peripherals/lcd/spi_lcd_touch).

--- a/components/lcd/esp_lcd_ili9341/esp_lcd_ili9341.c
+++ b/components/lcd/esp_lcd_ili9341/esp_lcd_ili9341.c
@@ -61,7 +61,7 @@ esp_err_t esp_lcd_new_panel_ili9341(const esp_lcd_panel_io_handle_t io, const es
         ESP_GOTO_ON_ERROR(gpio_config(&io_conf), err, TAG, "configure GPIO for RST line failed");
     }
 
-#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 4)
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
     switch (panel_dev_config->color_space) {
     case ESP_LCD_COLOR_SPACE_RGB:
         ili9341->madctl_val = 0;
@@ -74,15 +74,15 @@ esp_err_t esp_lcd_new_panel_ili9341(const esp_lcd_panel_io_handle_t io, const es
         break;
     }
 #else
-    switch (panel_dev_config->rgb_ele_order) {
-    case LCD_RGB_ELEMENT_ORDER_RGB:
+    switch (panel_dev_config->rgb_endian) {
+    case LCD_RGB_ENDIAN_RGB:
         ili9341->madctl_val = 0;
         break;
-    case LCD_RGB_ELEMENT_ORDER_BGR:
+    case LCD_RGB_ENDIAN_BGR:
         ili9341->madctl_val |= LCD_CMD_BGR_BIT;
         break;
     default:
-        ESP_GOTO_ON_FALSE(false, ESP_ERR_NOT_SUPPORTED, err, TAG, "unsupported color element order");
+        ESP_GOTO_ON_FALSE(false, ESP_ERR_NOT_SUPPORTED, err, TAG, "unsupported rgb endian");
         break;
     }
 #endif

--- a/components/lcd/esp_lcd_ili9341/idf_component.yml
+++ b/components/lcd/esp_lcd_ili9341/idf_component.yml
@@ -1,4 +1,4 @@
-version: "1.1.0"
+version: "1.2.0"
 description: ESP LCD ILI9341
 url: https://github.com/espressif/esp-bsp/tree/master/components/lcd/esp_lcd_ili9341
 dependencies:

--- a/components/lcd/esp_lcd_ili9341/include/esp_lcd_ili9341.h
+++ b/components/lcd/esp_lcd_ili9341/include/esp_lcd_ili9341.h
@@ -62,16 +62,17 @@ esp_err_t esp_lcd_new_panel_ili9341(const esp_lcd_panel_io_handle_t io, const es
  *
  * @param[in] sclk SPI clock pin number
  * @param[in] mosi SPI MOSI pin number
+ * @param[in] max_trans_sz Maximum transfer size in bytes
  *
  */
-#define ILI9341_PANEL_BUS_SPI_CONFIG(sclk, mosi)        \
-    {                                                   \
-        .sclk_io_num = sclk,                            \
-        .mosi_io_num = mosi,                            \
-        .miso_io_num = -1,                              \
-        .quadhd_io_num = -1,                            \
-        .quadwp_io_num = -1,                            \
-        .max_transfer_sz = SPI_LL_DMA_MAX_BIT_LEN >> 3, \
+#define ILI9341_PANEL_BUS_SPI_CONFIG(sclk, mosi, max_trans_sz)  \
+    {                                                           \
+        .sclk_io_num = sclk,                                    \
+        .mosi_io_num = mosi,                                    \
+        .miso_io_num = -1,                                      \
+        .quadhd_io_num = -1,                                    \
+        .quadwp_io_num = -1,                                    \
+        .max_transfer_sz = max_trans_sz,                        \
     }
 
 /**

--- a/components/lcd/esp_lcd_ili9341/test_apps/main/test_esp_lcd_ili9341.c
+++ b/components/lcd/esp_lcd_ili9341/test_apps/main/test_esp_lcd_ili9341.c
@@ -14,6 +14,7 @@
 #include "esp_heap_caps.h"
 #include "esp_log.h"
 #include "esp_lcd_panel_io_interface.h"
+#include "esp_lcd_panel_io.h"
 #include "esp_lcd_panel_ops.h"
 #include "unity.h"
 #include "unity_test_runner.h"
@@ -28,9 +29,17 @@
 #define TEST_PIN_NUM_LCD_CS         (GPIO_NUM_5)
 #define TEST_PIN_NUM_LCD_PCLK       (GPIO_NUM_7)
 #define TEST_PIN_NUM_LCD_DATA0      (GPIO_NUM_6)
-#define TEST_PIN_NUM_LCD_RST        (GPIO_NUM_48)
 #define TEST_PIN_NUM_LCD_DC         (GPIO_NUM_4)
+#if CONFIG_IDF_TARGET_ESP32S3
+#define TEST_PIN_NUM_LCD_RST        (GPIO_NUM_48)
+#else
+#define TEST_PIN_NUM_LCD_RST        (GPIO_NUM_1)
+#endif
+#if CONFIG_IDF_TARGET_ESP32S3
 #define TEST_PIN_NUM_LCD_BL         (GPIO_NUM_45)
+#else
+#define TEST_PIN_NUM_LCD_BL         (GPIO_NUM_0)
+#endif
 
 #define TEST_DELAY_TIME_MS          (3000)
 
@@ -82,7 +91,8 @@ TEST_CASE("test ili9341 to draw color bar with SPI interface", "[ili9341][spi]")
     gpio_set_level(TEST_PIN_NUM_LCD_BL, 1);
 
     ESP_LOGI(TAG, "Initialize SPI bus");
-    const spi_bus_config_t buscfg = ILI9341_PANEL_BUS_SPI_CONFIG(TEST_PIN_NUM_LCD_PCLK, TEST_PIN_NUM_LCD_DATA0);
+    const spi_bus_config_t buscfg = ILI9341_PANEL_BUS_SPI_CONFIG(TEST_PIN_NUM_LCD_PCLK, TEST_PIN_NUM_LCD_DATA0,
+                                    TEST_LCD_H_RES * 80 * TEST_LCD_BIT_PER_PIXEL / 8);
     TEST_ESP_OK(spi_bus_initialize(TEST_LCD_HOST, &buscfg, SPI_DMA_CH_AUTO));
 
     ESP_LOGI(TAG, "Install panel IO");
@@ -96,10 +106,10 @@ TEST_CASE("test ili9341 to draw color bar with SPI interface", "[ili9341][spi]")
     esp_lcd_panel_handle_t panel_handle = NULL;
     const esp_lcd_panel_dev_config_t panel_config = {
         .reset_gpio_num = TEST_PIN_NUM_LCD_RST, // Shared with Touch reset
-#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 4)
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
         .color_space = ESP_LCD_COLOR_SPACE_BGR,
 #else
-        .rgb_ele_order = LCD_RGB_ELEMENT_ORDER_BGR,
+        .rgb_endian = LCD_RGB_ENDIAN_BGR,
 #endif
         .bits_per_pixel = TEST_LCD_BIT_PER_PIXEL,
     };
@@ -107,10 +117,10 @@ TEST_CASE("test ili9341 to draw color bar with SPI interface", "[ili9341][spi]")
     TEST_ESP_OK(esp_lcd_panel_reset(panel_handle));
     TEST_ESP_OK(esp_lcd_panel_init(panel_handle));
     TEST_ESP_OK(esp_lcd_panel_mirror(panel_handle, true, true));
-#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)
-    TEST_ESP_OK(esp_lcd_panel_disp_on_off(panel_handle, true));
-#else
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
     TEST_ESP_OK(esp_lcd_panel_disp_off(panel_handle, false));
+#else
+    TEST_ESP_OK(esp_lcd_panel_disp_on_off(panel_handle, true));
 #endif
 
     test_draw_bitmap(panel_handle);

--- a/components/lcd/esp_lcd_st7796/README.md
+++ b/components/lcd/esp_lcd_st7796/README.md
@@ -60,10 +60,10 @@ Alternatively, you can create `idf_component.yml`. More is in [Espressif's docum
     // };
     const esp_lcd_panel_dev_config_t panel_config = {
         .reset_gpio_num = EXAMPLE_PIN_NUM_LCD_RST,      // Set to -1 if not use
-#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 4)      // Implemented by LCD command `36h`
-        .color_space = ESP_LCD_COLOR_SPACE_BGR,
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)      // Implemented by LCD command `36h`
+        .color_space = ESP_LCD_COLOR_SPACE_RGB,
 #else
-        .rgb_ele_order = LCD_RGB_ELEMENT_ORDER_BGR,
+        .rgb_endian = LCD_RGB_ENDIAN_RGB,
 #endif
         .bits_per_pixel = EXAMPLE_LCD_BIT_PER_PIXEL,    // Implemented by LCD command `3Ah` (16/18/24)
         // .vendor_config = &vendor_config,            // Uncomment this line if use custom initialization commands
@@ -71,5 +71,9 @@ Alternatively, you can create `idf_component.yml`. More is in [Espressif's docum
     ESP_ERROR_CHECK(esp_lcd_new_panel_st7796(io_handle, &panel_config, &panel_handle));
     ESP_ERROR_CHECK(esp_lcd_panel_reset(panel_handle));
     ESP_ERROR_CHECK(esp_lcd_panel_init(panel_handle));
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
+    ESP_ERROR_CHECK(esp_lcd_panel_disp_off(panel_handle, false));
+#else
     ESP_ERROR_CHECK(esp_lcd_panel_disp_on_off(panel_handle, true));
+#endif
 ```

--- a/components/lcd/esp_lcd_st7796/esp_lcd_st7796.c
+++ b/components/lcd/esp_lcd_st7796/esp_lcd_st7796.c
@@ -61,7 +61,7 @@ esp_err_t esp_lcd_new_panel_st7796(const esp_lcd_panel_io_handle_t io, const esp
         ESP_GOTO_ON_ERROR(gpio_config(&io_conf), err, TAG, "configure GPIO for RST line failed");
     }
 
-#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 4)
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
     switch (panel_dev_config->color_space) {
     case ESP_LCD_COLOR_SPACE_RGB:
         st7796->madctl_val = 0;
@@ -74,15 +74,15 @@ esp_err_t esp_lcd_new_panel_st7796(const esp_lcd_panel_io_handle_t io, const esp
         break;
     }
 #else
-    switch (panel_dev_config->rgb_ele_order) {
-    case LCD_RGB_ELEMENT_ORDER_RGB:
+    switch (panel_dev_config->rgb_endian) {
+    case LCD_RGB_ENDIAN_RGB:
         st7796->madctl_val = 0;
         break;
-    case LCD_RGB_ELEMENT_ORDER_BGR:
+    case LCD_RGB_ENDIAN_BGR:
         st7796->madctl_val |= LCD_CMD_BGR_BIT;
         break;
     default:
-        ESP_GOTO_ON_FALSE(false, ESP_ERR_NOT_SUPPORTED, err, TAG, "unsupported color element order");
+        ESP_GOTO_ON_FALSE(false, ESP_ERR_NOT_SUPPORTED, err, TAG, "unsupported rgb endian");
         break;
     }
 #endif

--- a/components/lcd/esp_lcd_st7796/idf_component.yml
+++ b/components/lcd/esp_lcd_st7796/idf_component.yml
@@ -1,4 +1,4 @@
-version: "1.1.0"
+version: "1.2.0"
 targets:
   - esp32s2
   - esp32s3

--- a/components/lcd/esp_lcd_st7796/include/esp_lcd_st7796.h
+++ b/components/lcd/esp_lcd_st7796/include/esp_lcd_st7796.h
@@ -67,6 +67,20 @@ esp_err_t esp_lcd_new_panel_st7796(const esp_lcd_panel_io_handle_t io, const esp
  * @param[in] d[0:15] I80 data pin number 0 ~ 15. Set to -1 if not used.
  *
  */
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(4, 4, 1)
+#define ST7796_PANEL_BUS_I80_CONFIG(max_trans_bytes, data_width, dc, wr, d0, d1, d2, d3, d4,    \
+                                    d5, d6, d7, d8 , d9, d10, d11, d12, d13, d14, d15)          \
+    {                                                                                           \
+        .clk_src = LCD_CLK_SRC_PLL160M,                                                         \
+        .dc_gpio_num = dc,                                                                      \
+        .wr_gpio_num = wr,                                                                      \
+        .data_gpio_nums = {                                                                     \
+            d0, d1, d2, d3, d4, d5, d6, d7, d8 , d9, d10, d11, d12, d13, d14, d15               \
+        },                                                                                      \
+        .bus_width = data_width,                                                                \
+        .max_transfer_bytes = max_trans_bytes,                                                  \
+    }
+#else
 #define ST7796_PANEL_BUS_I80_CONFIG(max_trans_bytes, data_width, dc, wr, d0, d1, d2, d3, d4,    \
                                     d5, d6, d7, d8 , d9, d10, d11, d12, d13, d14, d15)          \
     {                                                                                           \
@@ -81,6 +95,7 @@ esp_err_t esp_lcd_new_panel_st7796(const esp_lcd_panel_io_handle_t io, const esp
         .psram_trans_align = 64,                                                                \
         .sram_trans_align = 4,                                                                  \
     }
+#endif
 
 /**
  * @brief LCD panel IO configuration structure

--- a/components/lcd/esp_lcd_st7796/test_apps/main/idf_component.yml
+++ b/components/lcd/esp_lcd_st7796/test_apps/main/idf_component.yml
@@ -1,12 +1,6 @@
 ## IDF Component Manager Manifest File
 dependencies:
   idf: ">=4.4"
-  esp_lcd_panel_io_additions:
-    version: "^1"
-    public: true
-  esp_io_expander_tca9554:
-    version: "^1"
-    public: true
   esp_lcd_st7796:
     version: "*"
     override_path: "../../../esp_lcd_st7796"

--- a/components/lcd/esp_lcd_st7796/test_apps/main/test_esp_lcd_st7796.c
+++ b/components/lcd/esp_lcd_st7796/test_apps/main/test_esp_lcd_st7796.c
@@ -8,15 +8,12 @@
 
 #include "freertos/FreeRTOS.h"
 #include "freertos/task.h"
-#include "driver/i2c.h"
-#include "driver/spi_master.h"
+#include "freertos/semphr.h"
 #include "driver/gpio.h"
 #include "esp_heap_caps.h"
 #include "esp_log.h"
 #include "esp_lcd_panel_ops.h"
 #include "esp_lcd_panel_io.h"
-#include "esp_lcd_panel_io_additions.h"
-#include "esp_io_expander_tca9554.h"
 #include "unity.h"
 #include "unity_test_runner.h"
 
@@ -104,10 +101,10 @@ TEST_CASE("test st7796 to draw color bar with I80 interface", "[st7796][i80]")
     ESP_LOGI(TAG, "Install ST7796 panel driver");
     const esp_lcd_panel_dev_config_t panel_config = {
         .reset_gpio_num = TEST_PIN_NUM_LCD_RST,
-#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 4)
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
         .color_space = ESP_LCD_COLOR_SPACE_BGR,
 #else
-        .rgb_ele_order = LCD_RGB_ELEMENT_ORDER_BGR,
+        .rgb_endian = LCD_RGB_ENDIAN_BGR,
 #endif
         .bits_per_pixel = TEST_LCD_BIT_PER_PIXEL,
     };
@@ -115,7 +112,11 @@ TEST_CASE("test st7796 to draw color bar with I80 interface", "[st7796][i80]")
     TEST_ESP_OK(esp_lcd_new_panel_st7796(io_handle, &panel_config, &panel_handle));
     TEST_ESP_OK(esp_lcd_panel_reset(panel_handle));
     TEST_ESP_OK(esp_lcd_panel_init(panel_handle));
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
+    TEST_ESP_OK(esp_lcd_panel_disp_off(panel_handle, false));
+#else
     TEST_ESP_OK(esp_lcd_panel_disp_on_off(panel_handle, true));
+#endif
 
     test_draw_bitmap(panel_handle);
     vTaskDelay(pdMS_TO_TICKS(TEST_DELAY_TIME_MS));

